### PR TITLE
Add maxBounds, minZoom, and maxZoom to prism.json map config

### PR DESCRIPTION
**Pull request by reed1**
For issue #179

### CODE REVIEW

**Comment by wadhwamatic** [comment]
@reed1 - please add a deployment to test

**Comment by reed1** [comment]
@wadhwamatic , I deployed here: http://prism-r1.surge.sh/?hazardLayerId=hydra_s1&date=2022-04-16

with these map config:

```
"map": {
  "latitude": 12.058,
  "longitude": 105.281,
  "zoom": 6.49,
  "maxBounds": [101.070557,9.160099,108.837889,15.566551],
  "maxZoom": 9
},
```

**Comment by wadhwamatic** [comment]
@reed1 - this works, but I found there's a lot of trial and error to find this right configuration between the maxZoom and bounding box parameters. I'm wondering if there's a simpler way to do this, or a tool to help find the right configuration. Any suggestions?

**Comment by reed1** [comment]
@wadhwamatic maybe this will help, you can press Ctrl+B on map, the bounds and zoom will be printed on console

https://prism-debug-bounds.surge.sh/

image

**Comment by wadhwamatic** [comment]
> @wadhwamatic maybe this will help, you can press Ctrl+B on map, the bounds and zoom will be printed on console

@reed1 - this is helpful thanks. But, 1) it's not updating when you move around and if I refresh the page, the console doesn't log a new value, and 2) this isn't in the codebase, so we won't be able to use it beyond this temporary URL - please add a commit

**Comment by reed1** [comment]
@ericboucher @wadhwamatic I think this PR is safe to merge, please review.

For debugging boundaries, I created a new branch [feature/show-boundary-info-box](https://github.com/WFP-VAM/prism-frontend/tree/feature/show-boundary-info-box) as discussed before with Amit. There will be boundary info box on the bottom left of the map if REACT_APP_SHOW_BOUNDARY_INFO set to "true". You can test the feature here: https://prism-location-box.surge.sh/. If it's good, I can create a PR on that one.

image

**Comment by wadhwamatic** [comment]
Thanks @reed1 . The feature is useful. Please create a PR based on the branch you propose to merge (https://github.com/WFP-VAM/prism-frontend/tree/feature/show-boundary-info-box if I understood correctly)

**Comment by wadhwamatic** [comment]
@reed1 - please make an update to this PR so that we can move forward with a review 

cc @ericboucher

**Comment by reed1** [comment]
@wadhwamatic I put a separate PR here #525. I think this PR is fine as it is, just small additions that enables maxBounds, minZoom and maxZoom, or do you need anything else to update?

File: src/components/MapView/index.tsx [hunk 1]
```diff
@@ -419,6 +419,18 @@ function MapView({ classes }: MapViewProps) {
     setFirstSymbolId(layers?.find(layer => layer.type === 'symbol')?.id);
     dispatch(setMap(() => map));
     map.jumpTo({ center: [longitude, latitude], zoom });
+    const maxBounds = get(appConfig.map, 'maxBounds');
+    const minZoom = get(appConfig.map, 'minZoom');
+    const maxZoom = get(appConfig.map, 'maxZoom');
+    if (maxBounds) {
+      map.setMaxBounds(maxBounds);
```
Comment (ericboucher) [inline src/components/MapView/index.tsx:426]: what happens if you do setMaxBounds(undefined)?
could we do something like `map.setMaxBounds(maxBounds || undefined)`?

**Review by ericboucher — APPROVED** [review]
Looks good, just a minor comment to simplify the logic if possible but no big deal otherwise

**Pull request merged**